pkg/infra/iac2/templates/secret_version/factory.ts

```diff
@@ -10,17 +10,19 @@ interface Args {
 }
 
 // noinspection JSUnusedLocalSymbols
-function create(args: Args): aws.secretsmanager.SecretVersion {
-    return new aws.secretsmanager.SecretVersion(
-        args.SecretName,
-        {
-            secretId: args.Secret.id,
-            //TMPL {{ if eq .Type.Raw "string" }}
-            //TMPL secretString: fs.readFileSync(args.Path, 'utf-8').toString()
-            //TMPL {{ else }}
-            secretBinary: fs.readFileSync(args.Path, 'base64').toString(),
-            //TMPL {{ end }}
-        },
-        { protect: args.protect }
-    )
+function create(args: Args): void {
+    if (fs.existsSync(args.Path)) {
```

Comment:
this will cause failures if we return void since other things may depend on the secret. I ran into this earlier, so here we would regress and not be deployable

Reply:
It won't cause any failures because the dependency was for the policy which was erroneously pointing to the version's ARN instead of the secret's ARN. This is corrected in this PR (though I only fixed it for config, I'll go fix that for secret now) and the resulting index.ts has no compilation errors.

Reply:
well for rds we depend on the secret version and need that secret versions arn i thought? Or does it automatically work if you put the secrets arn?

Reply:
Yeah, the way it works is access is done at the `Secret` level and the `SecretVersion` is only used if you need something other than "latest" or to write a new version. In fact, looks like RDS already correctly uses secret:
```go
secretPolicyDoc := CreateAllowPolicyDocument([]string{"secretsmanager:GetSecretValue"}, []core.IaCValue{{Resource: secret, Property: ARN_IAC_VALUE}})
```